src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +55,1315 @@
     public String groupByTeam = null;
     public String showStats = null;
     public int startIndex;
+    private boolean isPanelsCollapsed;
+    
+    private FieldValidator validator = new FieldValidator();
 
     // used for html table ajax loading
     public String courseId = null;
     public String feedbackSessionName = null;
     public String ajaxStatus = null;
     public String sessionResultsHtmlTableAsString = null;
+    
 
+    // for question view
+    List<InstructorFeedbackResultsQuestionTable> questionPanels;
+    // for giver > question > recipient, recipient > question > giver...
+    LinkedHashMap<String, InstructorFeedbackResultsSectionPanel> sectionPanels;
+    
+    // TODO multiple page data classes inheriting this for each view type, 
+    // rather than an enum determining behavior in many methods
+    ViewType viewType;
+    enum ViewType {
+        QUESTION, GIVER_QUESTION_RECIPIENT, RECIPIENT_QUESTION_GIVER;
+        
+        public String toString() {
+            // replace _ to - to keep it consistent with old behavior
+            return name().toLowerCase().replaceAll("_", "-");
+        }
+        
+        public boolean isFirstGroupedByGiver() {
+            return this == GIVER_QUESTION_RECIPIENT;
+        }
+    }
+    
+    
     public InstructorFeedbackResultsPageData(AccountAttributes account) {
         super(account);
         startIndex = -1;
     }
+    
+    public void initForViewByQuestion(InstructorAttributes instructor, 
+                                      String selectedSection, String showStats, 
+                                      String groupByTeam) {
+        viewType = ViewType.QUESTION;
+        
+        this.instructor = instructor;
+        this.selectedSection = selectedSection;
+        this.showStats = showStats;
+        this.groupByTeam = groupByTeam;
+        
+        Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> questionToResponseMap = bundle.getQuestionResponseMap();
+        questionPanels = new ArrayList<InstructorFeedbackResultsQuestionTable>();
+        
+        for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> entry : questionToResponseMap.entrySet()) {
+            FeedbackQuestionAttributes question = entry.getKey();
+            List<FeedbackResponseAttributes> responses = entry.getValue();
+            
+            questionPanels.add(buildQuestionTableAndResponseRows(question, responses, ""));
```

Comment:
Should the signature for the `buildQuestionTableAndResponseRows(question, responses, "")` call here just eliminate the `""` in the parameter and place it in the method body?
It's only being called in one place (here) and it's already defaulting to `null` and `true` for the `participantIdentifier` and `isShowingResponseRows`.
Likewise for `buildQuestionTableWithoutResponseRows()`.